subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathInstrumentationIntegrationTest.groovy

```diff
@@ -144,6 +151,7 @@ class BuildScriptClasspathInstrumentationIntegrationTest extends AbstractIntegra
 
     def "order of entries in the effective classpath stays the same as in the original classpath"() {
         given:
+        requireOwnGradleUserHomeDir()
```

Comment:
Good idea! I can add a method with a "because" and deprecate the old method for now. Thinking about just deprecating it, since the method is used in lots of places and would increase the size of a review a lot. 

What do you think about that idea?

Reply:
Added `requireOwnGradleUserHomeDir(String because)` method